messaging/src/test/java/org/axonframework/messaging/MessageStreamTest.java

```diff
@@ -33,12 +33,13 @@
 /**
  * Test suite used to validate implementations of the {@link MessageStream}.
  *
- * @param <M> The type of {@link Message} contained in the {@link Entry entries} of the {@link MessageStream stream}
- *            under test.
+ * @param <IN>  The type of source {@link Message} contained in the {@link Entry entries} of the
+ *              {@link MessageStream stream} under test.
+ * @param <OUT> The type result {@link Message} after processing of the {@link MessageStream stream} under test.
  * @author Allard Buijze
  * @author Steven van Beelen
  */
-public abstract class MessageStreamTest<M extends Message<?>> {
+public abstract class MessageStreamTest<IN extends Message<?>, OUT extends Message<?>> {
```

Comment:
It was needed, because in case of the `IgnoredEntriesMessageStream` I have some entries as an input, but ignores them so the output have different type.

Reply:
Ah, it's necessary because `ignoredEntries` set's the generic of the `Message` to `Void`. 
To be honest, I don't think `ignoredEntries` has to do that. It is different from an `empty()` invocation, as it didn't start off as a void message stream. We force it to have no entries after all. 

To be more exact, I think we can keep the generics on the `ignoredEntries` method, and thus the `IgnoredEntriesMessageStream` instance. In doing so, I believe you wouldn't be required to specify the out going `Message` type in all the tests either.

But, perhaps I oversee something. So, what do you think, @MateuszNaKodach?

Reply:
Your totally right! I don't know why I was fixed on that Empty is not generic and the message needs to be void :P I'm introducing the generic.

Reply:
Done